### DISCUSSION TITLE
VRCNowPlaying - Show what media you're listening to via Chatbox OSC

### DISCUSSION THREAD

**Discussion by cyberkitsune** [vrchat-community/osc#136] [Show and tell]
### VRCNowPlaying - https://github.com/cyberkitsune/vrc-osc-scripts
Using Python and the Windows MediaManager API, I query what media you're listening to via a media player (Spotify, YouTube, Chrome Media API, etc...) and display it via the new Chatbubble via OSC! With this, others will be able to see what you're personally jamming out to!

I also utilize the `Cutlet` package if it is installed here to work around the lack of Unicode support via OSC, it will convert Japanese into romaji so the title doesn't appear as all question marks in VRChat.

You can find it in my [vrc-osc-scripts](https://github.com/cyberkitsune/vrc-osc-scripts) repo under the `VRCNowPlaying/` folder!

### Demonstration
![VRCNowPlaying in action!](https://raw.githubusercontent.com/cyberkitsune/vrc-osc-scripts/main/img/nowplaying.gif)

**Comment by EIA485** [comment]
nice! i made something similar in (rushed) c# [EIA485/oscChatBoxMusic](https://github.com/EIA485/oscChatBoxMusic)

**Comment by momo-the-monster** [comment]
@cyberkitsune - thanks for sharing! Your work has been added to [the official listing](https://github.com/vrchat-community/osc/wiki/Resources#misc).
